core/src/main/java/org/apache/carbondata/core/datastore/filesystem/AlluxioCarbonFile.java

```diff
@@ -94,14 +93,9 @@ public CarbonFile getParentFile() {
   public boolean renameForce(String changeToName) {
     FileSystem fs;
     try {
-      fs = fileStatus.getPath().getFileSystem(FileFactory.getConfiguration());
-      if (fs instanceof DistributedFileSystem) {
-        ((DistributedFileSystem) fs).rename(fileStatus.getPath(), new Path(changeToName),
-            org.apache.hadoop.fs.Options.Rename.OVERWRITE);
-        return true;
-      } else {
-        return false;
-      }
+      fs = fileStatus.getPath().getFileSystem(hadoopConf);
+      fs.delete(new Path(changeToName), true);
+      return fs.rename(fileStatus.getPath(), new Path(changeToName));
```

Comment:
It's a risky operation if 98 is failed then no way to get the file which is deleted in 97 line.  I think we should find a way to do it in single transaction.  Any better way @jackylk ?

Reply:
I am removing the delete..as kindle refer to the rename operation in Alluxio..it conflicts the way delete is present..which is handled in the alluxio version from 1.7.1

Reply:
You mean force rename is handled in Alluxio 1.7.1 ?  which Alluxio version are you using currently?

Reply:
Actually no I mean..it checks for the existence of the file and renames only ( else it will return false), but will not do a force rename. But I am thinking we can use the same alluxio fs copy, programatically to copy the same contents from one file to another file(changedName under the same directory).

Reply:
This problem is also for `S3CarbonFile` , `ViewFSCarbonFile`, `LocalCarbonFile`